app/components/CommentForm/index.js

```diff
@@ -1,153 +1,122 @@
 // @flow
 
-import { Component } from 'react';
-import { compose } from 'redux';
-import { connect } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import cx from 'classnames';
-import { getFormMeta, getFormValues, reduxForm, Field } from 'redux-form';
-import type { FormProps } from 'redux-form';
 import { EditorField } from 'app/components/Form';
 import Button from 'app/components/Button';
 import { ProfilePicture } from 'app/components/Image';
 import { addComment } from 'app/actions/CommentActions';
-import type { CommentEntity } from 'app/actions/CommentActions';
 import styles from './CommentForm.css';
 import DisplayContent from 'app/components/DisplayContent';
-
-// TODO: This can be removed if the editor importer gets an actual empty state.
-const EMPTY_STATE = '<p></p>';
-
-const validate = (values) => {
-  const errors = {};
-  if (!values.text) {
-    errors.text = 'Required';
-  }
-  return errors;
-};
+import { EDITOR_EMPTY } from 'app/utils/constants';
+import LegoFinalForm from 'app/components/Form/LegoFinalForm';
+import { Field } from 'react-final-form';
+import SubmissionError from 'app/components/Form/SubmissionError';
+import { createValidator, legoEditorRequired } from 'app/utils/validation';
 
 type Props = {
   contentTarget: string,
   user: Object,
   loggedIn: boolean,
-  addComment: (CommentEntity) => void,
-  parent: number,
-  submitText: string,
-  inlineMode: boolean,
-  initialized: boolean,
-  autoFocus: boolean,
-  isOpen: boolean,
-} & FormProps;
-
-class CommentForm extends Component<Props, { disabled: boolean }> {
-  constructor(props) {
-    super(props);
-    this.state = { disabled: !__CLIENT__ };
-  }
-  static defaultProps = {
-    submitText: 'Kommenter',
-    autoFocus: false,
-  };
+  submitText?: string,
+  inlineMode?: boolean,
+  autoFocus?: boolean,
+  parent?: number,
+};
 
-  onSubmit = ({ text }) => {
-    const { contentTarget, parent } = this.props;
-    this.props.addComment({
-      contentTarget,
-      text,
-      parent,
-    });
-  };
+const validate = createValidator({
+  text: [legoEditorRequired('Kommentaren kan ikke være tom')],
+});
 
-  enableForm = (e) => {
-    this.setState({ disabled: false });
-  };
+const CommentForm = ({
+  contentTarget,
+  user,
+  loggedIn,
+  submitText = 'Kommenter',
+  inlineMode = false,
+  autoFocus = false,
+  parent,
+}: Props) => {
+  const dispatch = useDispatch();
+  const [disabled, setDisabled] = useState(true);
 
-  render() {
-    const {
-      handleSubmit,
-      pristine,
-      submitting,
-      user,
-      isOpen,
-      loggedIn,
-      submitText,
-      inlineMode,
-      autoFocus,
-      initialized,
-    } = this.props;
-    const className = inlineMode ? styles.inlineForm : styles.form;
+  useEffect(() => {
+    // Workaround to make sure we re-render editor in enabled state on client after ssr
+    setDisabled(false);
+  }, []);
```

Comment:
Haha, you were the one who added it. [In this pr](https://github.com/webkom/lego-webapp/commit/4e91800f106ab3235db25a9ce9366537baaf48f2). I just simplified it a bit, by enabling the form immediately on the client instead of waiting for a hover/click event.

 If I remove it, the comment input lego-editor is always disabled when SSRed. I don't really understand why, but hopefully your pr makes sense to you.